src/OmenThumbnailMapping.sol

```diff
@@ -0,0 +1,45 @@
+// SPDX-License-Identifier: GPL-3.0
+pragma solidity ^0.8.24;
+
+import {IERC20} from "../interfaces/IERC20.sol";
+
+contract OmenThumbnailMapping {
+    // Mapping from market's address to image's IPFS hash
+    mapping(address => bytes32) private marketAddressToIPFSHash;
+    mapping(address => address) private marketAddressToLatestImageChanger;
+
+    function get(address marketAddress) public view returns (bytes32) {
+        // Get IPFS hash of thumbnail for the given market.
+        return marketAddressToIPFSHash[marketAddress];
+    }
+
+    function set(address marketAddress, bytes32 image_hash) public requireThatSenderCanChangeImage(marketAddress) {
+        // Update IPFS hash of thumbnail for the given market.
+        marketAddressToIPFSHash[marketAddress] = image_hash;
+        marketAddressToLatestImageChanger[marketAddress] = msg.sender;
+    }
+
+    function remove(address marketAddress) public requireThatSenderCanChangeImage(marketAddress) {
+        // Remove IPFS hash of thumbnail for the given market.
+        delete marketAddressToIPFSHash[marketAddress];
+        marketAddressToLatestImageChanger[marketAddress] = msg.sender;
```

Comment:
Saving only the `msg.sender` in the mapping prevents us from differentiating between `set`,  and `remove` ops. Do we need this differentiation?
This is worth clarifying because the contract will be immutable after deploying.

Reply:
I guess no? Or do you think we do? Even with `set` one can set it to NULL_BYTES and then it's the same as remove, isn't it?

Reply:
My guess is also no and agree that update/delete difference is blurry.
Hence no point in differentiating set/remove.